src/languageFacts/scssFacts.ts

```diff
@@ -0,0 +1,760 @@
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/string' }],
+		kind: CompletionItemKind.Module
+	},
+	{
+		label: 'sass:color',
+		documentation: localize(
+			'scss.builtin.sass:color',
+			'Generates new colors based on existing ones, making it easy to build color themes.'
+		),
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/color' }],
+		kind: CompletionItemKind.Module
+	},
+	{
+		label: 'sass:list',
+		documentation: localize('scss.builtin.sass:list', 'Lets you access and modify values in lists.'),
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/list' }],
+		kind: CompletionItemKind.Module
+	},
+	{
+		label: 'sass:map',
+		documentation: localize(
+			'scss.builtin.sass:map',
+			'Makes it possible to look up the value associated with a key in a map, and much more.'
+		),
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/map' }],
+		kind: CompletionItemKind.Module
+	},
+	{
+		label: 'sass:selector',
+		documentation: localize('scss.builtin.sass:selector', 'Provides access to Sass’s powerful selector engine.'),
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/selector' }],
+		kind: CompletionItemKind.Module
+	},
+	{
+		label: 'sass:meta',
+		documentation: localize('scss.builtin.sass:meta', 'Exposes the details of Sass’s inner workings.'),
+		references: [{ name: 'Sass documentation', url: 'https://sass-lang.com/documentation/modules/meta' }],
+		kind: CompletionItemKind.Module
+	}
+];
+
+const variableDefaults: { [key: string]: string } = {
+	$red: '1',
+	$green: '2',
+	$blue: '3',
+	$alpha: '1.0',
+	$color: '#000000',
+	$weight: '0.5',
+	$hue: '0',
+	$saturation: '0%',
+	$lightness: '0%',
+	$degrees: '0',
+	$amount: '0',
+	$string: '""',
+	$substring: '"s"',
+	$number: '0',
+	$limit: '1'
+};
+
+function createReplaceFunction() {
+	let tabStopCounter = 1;
+	return (_match: string, p1: string) => {
+		return '\\' + p1 + ': ${' + tabStopCounter++ + ':' + (variableDefaults[p1] || '') + '}';
+	};
+}
+
+function functionToCompletionItem(proposal: IFunctionInfo): CompletionItem {
+	const label = proposal.func.substr(0, proposal.func.indexOf('('));
+	const insertText = proposal.func.replace(/\[?(\$\w+)\]?/g, createReplaceFunction());
+
+	const item: CompletionItem = {
+		label: label,
+		detail: proposal.func,
+		documentation: proposal.desc,
+		insertText,
+		insertTextFormat: InsertTextFormat.Snippet,
+		kind: CompletionItemKind.Function
+	};
+
+	return item;
+}
+
+class SCSSDataManager {
+	private _atDirectives: CompletionItem[];
+	private _moduleAtDirectives: CompletionItem[];
+	private _builtinModules: CompletionItem[];
+
+	constructor() {
+		this._atDirectives = atDirectives.map(i => {
+			return {
```

Comment:
If I get this right, there's no need to create a clone here.
Just do a `atDirectives.forEach(addReferenceDocumentation)`

Reply:
It's necessary because the `CompletionItem` returned are later modified with additional `sortText` and `TextEdit`, so I create a clone here.